### ISSUE TITLE
AL05 and unparseable error when inserting hard-coded values into table MariaDB

### ISSUE THREAD

**Issue by cooper-grill** [sqlfluff/sqlfluff#7160]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Incorrect alias formatting and syntax error on (what I think are called?) table value constructors

### Expected Behaviour

For the provided query to be left unchanged

### Observed Behaviour

```
L:  20 | P:   6 | AL05 | Alias 'temp' is never used in SELECT statement.
                       | [aliasing.unused]
L:  20 | P:  10 |  PRS | Line 20, Position 10: Found unparsable section:
                       | '(number, part, text) WHERE NOT EXISTS (\n...'
```

### How to reproduce

run `sqlfluff fix --config .sqlfluff --show-lint-violations <sql file>` on the below content:

```
INSERT INTO test
(col1, col2)
SELECT *
FROM (
    VALUES
    (1, "a"),
    (2, "b")
) AS temp(col1, col2) WHERE NOT EXISTS (
        SELECT NULL FROM test
    );

```

### Dialect

mariadb

### Version

3.4.2

### Configuration

```
[sqlfluff]
dialect = mariadb
exclude_rules = CV08, LT14, RF01, RF02, RF04, RF05
fix_even_unparsable = True
large_file_skip_byte_limit = 0
max_line_length = 120

[sqlfluff:indentation]
allow_implicit_indents = True

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by cooper-grill** [comment]
I believe this might also be related/similar to #7039 and #6733

**Comment by keraion** [comment]
I understand the parsing issue; however, the example is not using the `temp` alias which would seemingly be correct with triggering AL05. I'm assuming that the alias is not explicitly required for the statement to run. If that's not the case, please let me know.

Something that would not trigger AL05 would be the following:
```sql
INSERT INTO test
(col1, col2)
SELECT *
FROM (
    VALUES
    (1, "a"),
    (2, "b")
) AS temp(col1, col2) WHERE NOT EXISTS (
        SELECT NULL FROM test WHERE temp.col1 = test.col1
    );
```

**Comment by keraion** [comment]
I see [documentation from mysql](https://dev.mysql.com/doc/refman/8.4/en/derived-tables.html) that would suggest that the alias is required. I'll assume it would be the same with mariadb and get something together for this.
